src/server/ApiRoutes/Search.js

```diff
@@ -68,34 +69,48 @@ function fetchResults(searchKeywords = '', page, sortBy, order, field, filters,
       .catch(console.error))
     .then((response) => {
       const [results, aggregations, drbbResults] = response;
-      return Promise.all(results.itemListElement.map((resultObj) => {
+      const locationCodes = new Set();
+      const { itemListElement } = results;
+      itemListElement.forEach((resultObj) => {
+        let numItemsProcessed = 0;
         const { result } = resultObj;
-        return addLocationUrls(result).then((updatedResult) => {
-          const { holdings } = updatedResult;
-          if (!holdings) return { ...resultObj, updatedResult };
-          addCheckInItems(updatedResult);
-          holdings.slice(0, itemTableLimit).forEach(
-            holding => addHoldingDefinition(holding));
-          return { ...resultObj, updatedResult };
+        const { holdings } = resultObj.result;
+        if (holdings) {
+          addCheckInItems(result);
+          holdings.slice(0, itemTableLimit).forEach((holding) => {
+            addHoldingDefinition(holding);
+            if (numItemsProcessed < 0 && holding.location) {
```

Comment:
`numItemsProcessed` starts at `0` and is only ever increased as far as I can tell. Should the check really be `numItemsProcessed < 0` ?

Reply:
That is a typo... It is correct for the bib version of this logic and that is why it still worked. Pushing the correction right now.